pandas/core/generic.py

```diff
@@ -9236,6 +9237,7 @@ def where(
         name="mask",
         name_other="where",
     )
+    @deprecate_nonkeyword_arguments(version="2.0", allowed_args=["self", "cond"])
```

Comment:
let's make this `version=None`

Reply:
also, let's allow `other` to be positional as well, just like in `where` https://github.com/pandas-dev/pandas/pull/41523/files